be/src/storage/tablet_updates.cpp

```diff
@@ -2090,6 +2096,11 @@ Status TabletUpdates::compaction(MemTracker* mem_tracker) {
                                         new_bytes > compaction_result_bytes_threashold * 3 / 2)) {
             break;
         }
+        // Partial update generate empty rowset, compact them first.
+        // Or partial update by column will trigger too many useless compaction cost.
+        if (info->inputs.size() > 1 && has_partial_update_by_column) {
```

Comment:
If just one rowset is empty, the strategy will be triggered?

Reply:
No, at least two rowsets.